app/components/provider_user_notification_preferences_component.rb

```diff
@@ -0,0 +1,28 @@
+class ProviderUserNotificationPreferencesComponent < ViewComponent::Base
+  include ViewHelper
+  attr_reader :notification_preferences
+
+  CUSTOM_LEGENDS = {
+    application_withdrawn: 'Application withdrawn by candidate',
```

Comment:
I think we might need to define this for all notifications to avoid relying on the column name incase it might be renamed, also what do you think about moving those to locales instead?